packages/Microsoft.Bot.Components.Telephony/Actions/CallTransferDialog.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    public class CallTransferDialog : Dialog
+    {
+        [JsonConstructor]
+        public CallTransferDialog([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        [JsonProperty("$kind")]
+        public const string Kind = "CallTransferDialog";
+
+        [JsonProperty("targetPhoneNumber")]
+        public StringExpression TargetPhoneNumber { get; set; }
+
+        public async override Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var targetPhoneNumber = TargetPhoneNumber.GetValue(dc.State);
+
+            await dc.Context.SendActivityAsync($"Transferring to \"{targetPhoneNumber}\"...");
```

Comment:
Cannot hardcode strings

Reply:
what's the alternative?

Reply:
Check this out: https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs#L105

Allows authors to specify the activity that gets sent, including whether its a card or text, whether it comes from the language generation system (and thus getitng free localization), etc

Reply:
This should be optional also, since some may not want to send an activity.